board: stm32_min_dev: Add support for blue/black variant

sidcha opened the pull request:
This patch adds support for two variants of stm32_min_dev board configuration.

This patch reverts changes made in [c50a88] and [fa3af7] while still retaining the change.

sidcha committed:
board: stm32_min_dev: Add support for blue/black variant

Add support for blue pill and black pill variants of the STM32
minimum development board by splitting board configuration into
stm32_min_dev_blue and stm32_min_dev_black.

CODEOWNERS: Add myself (@cbsiddharth) as codeowner for stm32_min_dev

Signed-off-by: Siddharth Chandrasekaran <siddharth@embedjournal.com>

sidcha committed:
doc: stm32_min_dev: Update for blue/black variant support

Replace image and reword some sections to indicate the presence of two
board variants: stm32_min_dev_(blue|black)

Signed-off-by: Siddharth Chandrasekaran <siddharth@embedjournal.com>

dbkinder reviewed (approved):
LGTM, thanks.

erwango commented on boards/arm/stm32_min_dev/stm32_min_dev_black_defconfig:
```diff
@@ -0,0 +1,37 @@
+CONFIG_ARM=y
```
Miss
```
# SPDX-License-Identifier: Apache-2.0
#
```

cbsiddharth replied:
Added

erwango reviewed (commented):
Approach looks sane.
Since difference is located in dts but minimal, have you tried to have a common dtsi file that would be included by both variants and overwritten with the differences?

galak commented:
Looks like you need to add these platforms to the exclude lists in `tests/kernel/interrupt/testcase.yaml` & `tests/kernel/gen_isr_table/testcase.yaml`

cbsiddharth commented:
@galak, shall I make that as a separate commit or should I rebase and fixup?

sidcha committed:
tests: kernel: Exclude platforms stm32_min_dev_*

Exclude the two variants of smt32_min_dev (stm32_min_dev_black and
stm32_min_dev_blue) from kernel tests.

Signed-off-by: Siddharth Chandrasekaran <siddharth@embedjournal.com>

galak commented:
> @galak, shall I make that as a separate commit or should I rebase and fixup?

a separate commit in the PR is fine.

cbsiddharth commented:
> a separate commit in the PR is fine.
Done.

explora26 commented on boards/arm/stm32_min_dev/stm32_min_dev.dtsi:
```diff
@@ -70,7 +70,3 @@
-
-&usb {
-	status = "ok";
-};
```
Both black and blue variant support USB device, Is there any reason to remove this ?
